server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessagesMethodTest.java

```diff
@@ -331,4 +331,34 @@ public void getMessagesShouldReturnMandatoryPropertiesMessagesWhenNotAsked() thr
             .body(ARGUMENTS + ".list[0].id", equalTo("username@domain.tld|mailbox|1"))
             .body(ARGUMENTS + ".list[0].subject", equalTo("my test subject"));
     }
+
+    @Test
+    public void getMessagesShouldReturnAttachmentsWhenSome() throws Exception {
+        jmapServer.serverProbe().createMailbox(MailboxConstants.USER_NAMESPACE, username, "inbox");
+
+        ZonedDateTime dateTime = ZonedDateTime.parse("2014-10-30T14:12:00Z");
+        jmapServer.serverProbe().appendMessage(username, new MailboxPath(MailboxConstants.USER_NAMESPACE, username, "inbox"),
+                ClassLoader.getSystemResourceAsStream("twoAttachments.eml"), Date.from(dateTime.toInstant()), false, new Flags());
+        
+        await();
+        
+        given()
+            .accept(ContentType.JSON)
+            .contentType(ContentType.JSON)
+            .header("Authorization", accessToken.serialize())
+            .body("[[\"getMessages\", {\"ids\": [\"" + username + "|inbox|1\"]}, \"#0\"]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("messages"))
+            .body(ARGUMENTS + ".list", hasSize(1))
+            .body(ARGUMENTS + ".list[0].attachments", hasSize(2))
+            .body(ARGUMENTS + ".list[0].attachments[0].blobId", equalTo("223a76c0e8c1b1762487d8e0598bd88497d73ef2"))
+            .body(ARGUMENTS + ".list[0].attachments[0].type", equalTo("image/jpeg"))
```

Comment:
:-( I am sure the UX team would love to have the file name for display, WDYT ?

I just check, it is part of the JMAP spec, and it is optional, but definitly cool